data_model/models.py

```diff
@@ -0,0 +1,108 @@
+    # StringsInfo (data), Triage (permalink), UnpacMe (permalink), XlmMacroDeobfuscator,
+    # Yara (report.list_el.url/rule_url), Yaraify (link),
+    # HybridAnalysisFileAnalyzer (domains),
+    # VirusTotalV3FileAnalyzer (data.relationships.contacted_urls/contacted_domains)
+    related_threats = pg_fields.ArrayField(
+        models.CharField(max_length=100), null=True
+    )  # threats/related_threats
+    # GoogleSafeBrowsing, QuarkEngineAPK (crimes.crime)
+    malware_family = models.CharField(
+        max_length=100, null=True
+    )  # family/family_name/malware_family
+    # HybridAnalysisObservable, Intezer (family_name), Cuckoo, MwDB,
+    # Triage (analysis.family), UnpacMe (results.malware_id.malware_family),
+    # VirusTotalV3FileAnalyzer
+    # (attributes.last_analysis_results.list_el.results/attributes.names)
+    additional_info = (
+        models.JSONField()
+    )  # field for additional information related to a specific analyzer
+
+
+class DomainDataModel(BaseDataModel):
+    ietf_report = models.ForeignKey(
+        IETFReport, on_delete=models.CASCADE, null=True
+    )  # pdns
+    rank = models.IntegerField(null=True)  # Tranco
+
+
+class IPDataModel(BaseDataModel):
+    ietf_report = models.ForeignKey(
+        IETFReport, on_delete=models.CASCADE, null=True
+    )  # pdns
+    asn = models.IntegerField(null=True)  # BGPRanking, MaxMind
+    asn_rank = models.DecimalField(null=True)  # BGPRanking
+    certificates = models.JSONField(null=True)  # CIRCL_PSSL
+    org_name = models.CharField(max_length=100, null=True)  # GreyNoise
+    country = models.CharField(max_length=100, null=True)  # MaxMind, AbuseIPDB
+    country_code = models.CharField(max_length=100, null=True)  # MaxMind, AbuseIPDB
+    registered_country = models.CharField(
+        max_length=100, null=True
+    )  # MaxMind, AbuseIPDB
+    registered_country_code = models.CharField(
+        max_length=100, null=True
+    )  # MaxMind, AbuseIPDB
+    isp = models.CharField(max_length=100, null=True)  # AbuseIPDB
+    is_anonymizer = models.BooleanField(null=True)  # TorProject, Crowdsec
+    is_tor_exit_node = models.BooleanField(null=True)  # TorProject, Crowdsec
+    # additional_info
+    # behavior = models.CharField(max_length=100, null=True)  # Crowdsec
+    # noise = models.BooleanField(null=True)  # GreyNoise
+    # riot = models.BooleanField(null=True)  # GreyNoise
+
+
+class FileDataModel(BaseDataModel):
+    tags = pg_fields.ArrayField(
+        models.CharField(max_length=100), null=True
+    )  # HybridAnalysisFileAnalyzer, MalwareBazaarFileAnalyzer, MwDB,
+    # VirusTotalV3FileAnalyzer (report.data.tags)
+    signatures = pg_fields.ArrayField(
+        models.CharField(max_length=100), null=True
+    )  # ClamAvFileAnalyzer, MalwareBazaarFileAnalyzer, Yara (report.list_el.match)
+    yara_rules = models.JSONField(null=True)  # MalwareBazaarFileAnalyzer,
+    # Yaraify (report.data.tasks.static_result)
```

Comment:
I would do something like this 
```python3
enum.py
class SignaturesChoices(Choices):
      CLAMAV = "ClamAv"
      SIGMA = "Sigma"
      Yara = "Yara"
      ...
file: models.py
class Signature(models.Model):
    name = models.CharField(max_length=100) 
    SIGNATURES = SignaturesChoices
    signature = models.JsonField()

class FileDataModel(BaseDataModel):
     ...
     signatures = models.ManyToMany(Signature)
```

In this way you can
- find ALL report that matched a specific signature
- find all signatures of a provider in a specific report
- find all signatures in a specific report
- do not have different fields that logically means the same thing

@mlodic could this be a valid compromise?